pkg/apis/extensions/v1beta1/register.go

```diff
@@ -60,6 +58,7 @@ func addKnownTypes() {
 		&ListOptions{},
 		&ConfigMap{},
 		&ConfigMapList{},
+		&v1.DeleteOptions{},
```

Comment:
Opposite argument.  Every group should have a DeleteOptions kind
registered to it that converts to the internal DeleteOptions.  When we
rev delete options, we would also update those groups if necessary.

The go types are just schemas.  Saying "extensions/v1beta1" used the
core group "v1" delete options is all those code is doing (vs copying
the struct for no reason).

Everything in unversioned should really just be a common API group
that other groups are required to select a version of.  There is no
such thing as unversioned - there is just "things we haven't versioned
yet" and "things retrieved outside a version concept that we must be
100% back compatible on"